src/demo-app/demo-app/demo-app.ts

```diff
@@ -17,8 +26,12 @@ export class Home {}
   templateUrl: 'demo-app.html',
   styleUrls: ['demo-app.css'],
   encapsulation: ViewEncapsulation.None,
+  changeDetection: isPushChangeDetection ?
+    ChangeDetectionStrategy.OnPush :
+    ChangeDetectionStrategy.Default
```

Comment:
True. Yeah I initially thought that `setItem(key, changeDetectionStrategy)` would work, but it maps to integers.